analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/NonNullableValueTypeCheck.cs

```diff
@@ -24,9 +24,21 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal sealed class NonNullableValueTypeCheck : SymbolicCheck
     {
-        protected override ProgramState PostProcessSimple(SymbolicContext context) =>
-            context.Operation.Instance.Type.IsNonNullableValueType()
-                ? context.SetOperationConstraint(ObjectConstraint.NotNull)
-                : context.State;
+        protected override ProgramState PostProcessSimple(SymbolicContext context)
+        {
+            var state = context.State;
+            var operation = context.Operation.Instance;
+            if (operation.Type is { } type && type.IsNonNullableValueType())
```

Comment:
This will have a small rebase conflict after #6993, right? You could have rebased on that one.